Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Registration.hs

```diff
@@ -127,3 +182,22 @@ buildRegistrationToken personId merchantId = do
         merchantId = merchantId,
         createdAt = now
       }
+
+genTOTP :: Text -> IO OTP
+genTOTP secretKey = do
+  time <- DT.getPOSIXTime
+  let secretKeyBytes = case Base32.decodeBase32 (T.encodeUtf8 secretKey) of
+        Left _ -> error "Decoding error: "
+        Right bytes -> bytes
+      code = totp defaultTOTPParams secretKeyBytes (floor time)
```

Comment:
Directly return `totp defaultTOTPParams secretKeyBytes (floor time)` instead of storing in code variable and then returning.